python/cudf/cudf/core/column/categorical.py

```diff
@@ -77,34 +105,34 @@ def __init__(self, column, parent=None):
             raise AttributeError(
                 "Can only use .cat accessor with a 'category' dtype"
             )
-        self._column = column
-        self._parent = parent
+        super().__init__(column=column, parent=parent)
 
     @property
-    def categories(self):
+    def categories(self) -> "cudf.Index":
         """
         The categories of this categorical.
         """
         return cudf.core.index.as_index(self._column.categories)
 
     @property
-    def codes(self):
+    def codes(self) -> "cudf.Series":
         """
         Return Series of codes as well as the index.
         """
+        assert self._parent is None or isinstance(self._parent, cudf.Series)
```

Comment:
Leftover debugging?

Reply:
Not leftover debugging, points to a broken function which can only handle where `self._parent` is a specific type of value.

Reply:
Should fix this somehow that doesn't require an `assert`.